website/docs: Reorganize policy documentation (#21133)
goauthentik/authentik: website/docs/customize/policies/bindings.md

---
title: Policy Bindings and Evaluation
tags:
    - policy
    - bindings
    - access-control
---

> For step-by-step instructions on creating and attaching policies, see [Working with policies](./working_with_policies.md). This page focuses on where policy bindings apply, how authentik evaluates them, and which options affect the result.

Reviewer: This is a good addition... linking from the conceptual to the procedural.

- `All`: the target passes only when every binding passes

The default mode is `Any`.

### Order

Bindings are evaluated in ascending order. This matters most when you inspect logs or when you are combining multiple policies that produce end-user messages.

### Enabled bindings only

Disabled bindings are skipped entirely.

### Direct user and group bindings

Bindings to users and groups are evaluated alongside policy bindings:

- a user binding passes when the current user matches
- a group binding passes when the current user is a member of that group

These are a simple way to mix static membership checks with policy-based logic.

### No bindings means pass

If a target has no applicable bindings, authentik treats the result as passing.

## Binding options

Bindings have a few important options beyond the target and order.

### Negate

**Negate** flips the pass or fail result of the binding. This is useful when you want to express "everyone except this group" or turn an allow-style policy into a deny-style rule without copying it.

Negation only changes the boolean result. Any messages returned by the policy are left unchanged.

### Timeout

> **Timeout** limits how long authentik will wait for a policy execution before it is terminated. This is especially relevant for expression policies or other policy types that may call external systems.

Reviewer 2: ```suggestion
**Timeout** limits how long authentik will wait for a policy execution before it is terminated. This is especially relevant for expression policies or other policy types that may call external systems. Defaults to 30 seconds.
```

### Failure result

If a policy errors during execution, **Failure result** decides whether authentik should treat that failure as pass or fail.

Use this carefully:

- fail closed when the policy protects access to something sensitive
- fail open only when availability is more important than enforcement for that specific check

### Execution logging

> Individual policies can enable **Execution logging**. When enabled, authentik logs every execution of that policy, not only failures and exceptions. This is helpful while debugging complex access rules.

Reviewer 2: ```suggestion
Individual policies can enable **Execution logging**. When enabled, authentik logs every execution of that policy, not only failures and exceptions. This is helpful when debugging complex access rules.
```

## Common patterns

> ### Gate an application with a group or policy

Reviewer 2: ```suggestion
### Limit application access with a group or policy
```
